lib/connect2.lua

```diff
@@ -0,0 +1,155 @@
+local Roact = require(script.Parent.Parent.Roact)
+local storeKey = require(script.Parent.storeKey)
+local shallowEqual = require(script.Parent.shallowEqual)
+local join = require(script.Parent.join)
+
+--[[
+	Formats a multi-line message with printf-style placeholders.
+]]
+local function formatMessage(lines, parameters)
+	return table.concat(lines, "\n"):format(unpack(parameters or {}))
+end
+
+local function noop()
+	return nil
+end
+
+--[[
+	mapStateToProps: (storeState, props) -> partialProps
+	mapDispatchToProps: (dispatch) -> partialProps
+]]
+local function connect(mapStateToProps, mapDispatchToProps)
+	local connectTrace = debug.traceback()
+
+	if mapStateToProps ~= nil then
+		assert(typeof(mapStateToProps) == "function", "mapStateToProps must be a function or nil!")
+	else
+		mapStateToProps = noop
+	end
+
+	if mapDispatchToProps ~= nil then
+		assert(typeof(mapDispatchToProps) == "function", "mapDispatchToProps must be a function or nil!")
+	else
+		mapDispatchToProps = noop
+	end
+
+	return function(innerComponent)
+		if innerComponent == nil then
+			local message = formatMessage({
+				"connect returns a function that must be passed a component.",
+				"Check the connection at:",
+				"%s",
+			}, {
+				connectTrace,
+			})
+
+			error(message, 0)
```

Comment:
It removes the prefix on your error message that points to a line in the source as the blame. It doesn't affect the stack trace, but is useful when the line where the error occurred isn't the place that we want to highlight.

I suspect this case could actually be `2` instead of `0` to point at the caller!

Reply:
Pointing to  the caller should work here, yeah.  I think in most cases it's not going to be difficult to track down this error, but more specific is probably better.